addons/l10n_account_withholding/tests/test_account_withholding.py

```diff
@@ -0,0 +1,789 @@
+        tax.invoice_repartition_line_ids.filtered(lambda x: x.repartition_type == 'base').tag_ids = base_tag or self.env['account.account.tag'].create({
+            'name': f'Base Tag {name}',
+            'applicability': 'taxes',
+        })
+        return tax
+
+    def _get_tax_tag(self, tax):
+        return {
+            'tax': tax.invoice_repartition_line_ids.filtered(lambda x: x.repartition_type == 'tax').tag_ids.ids,
+            'base': tax.invoice_repartition_line_ids.filtered(lambda x: x.repartition_type == 'base').tag_ids.ids,
+        }
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        # Set the withholding account
+        cls.company_data['company'].withholding_tax_base_account_id = cls.env['account.account'].create({
+            'code': 'WITHB',
+            'name': 'Withholding Tax Base Account',
+            'reconcile': True,
+            'account_type': 'asset_current',
+        })
+        # Create a sequence to set on tax C
+        # Note that we use no gap for the sake of testing, in order to have the sequences be rollbacked between tests.
+        cls.withholding_sequence = cls.env['ir.sequence'].create({
+            'implementation': 'no_gap',
+            'name': 'Withholding Sequence',
+            'padding': 4,
+            'number_increment': 1,
+        })
+        # Prepare two withholding taxes.
+        cls.tax_sale_b = cls._setup_tax(cls, 'Withholding Tax', 1)
+        cls.tax_sale_c = cls._setup_tax(cls, 'Withholding Tax 2', 2, cls.withholding_sequence)
+        # Prepare one draft invoice for the tests.
+        cls.invoice = cls.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'date': '2024-01-01',
+            'invoice_date': '2024-01-01',
+            'partner_id': cls.partner_a.id,
+            'invoice_line_ids': [Command.create({'product_id': cls.product_a.id, 'price_unit': 1000.0})],
+        })
+        # Set a withholding tax on product B to test later.
+        cls.product_b.taxes_id = cls.tax_sale_b
+        # We'll need a foreign currency
+        cls.foreign_currency = cls.setup_other_currency('EUR')
+        # Fiscal position
+        cls.fiscal_pos_withh = cls.env['account.fiscal.position'].create({
+            'name': 'fiscal_pos_withh',
+            'tax_ids': ([(0, None, {'tax_src_id': cls.tax_sale_b.id, 'tax_dest_id': cls.tax_sale_c.id})]),
+        })
+        # Outstanding account
+        cls.outstanding_account = cls.env['account.account'].create({
+            'name': "Outstanding Payments",
+            'code': 'OSTP420',
+            'reconcile': False,  # On purpose for testing.
+            'account_type': 'asset_current'
+        })
+        # Second tax sale account for cases where we want multiple repartition lines
+        cls.tax_sale_account = cls.company_data['default_account_tax_sale'].copy()
+
+    def _register_payment(self, create_vals=None, enable_withholding=False, with_default_line=False):
```

Comment:
To me such helpers are not helping imo... just create what you need in the test, without building a huge spaguetti of sub methods that do magic things^^ It's way cleaner and understanding imo